contracts/FleetFactoryDeterministic.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity >= 0.5.0 < 0.7.0;
+
+import "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxyFactory.sol";
+import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+
+contract FleetFactoryDeterministic {
+  GnosisSafeProxyFactory public proxyFactory;
+
+  event FleetDeployed(address indexed owner, address[] fleet);
+
+  constructor(GnosisSafeProxyFactory _proxyFactory) public {
+    proxyFactory = _proxyFactory;
+  }
+
+  function deployFleetWithNonce(address owner, uint256 size, address template, uint256 saltNonce) external {
+    GnosisSafeProxyFactory _proxyFactory = proxyFactory;
+    address[] memory fleet = new address[](size);
+    address[] memory ownerList = new address[](1);
+    ownerList[0] = owner;
+    for (uint i = 0; i < size; i++) {
+      address payable proxy = address(
+        _proxyFactory.createProxyWithNonce(
+          template,
+          "",
+          uint256(
+            keccak256(
+              abi.encodePacked(saltNonce, i)
+            )
+          )
+        )
+      );
+      fleet[i] = proxy;
```

Comment:
maybe check that `proxy != address(0)` as the deployment failed in that case (should normally not happen ... most likely case is that a contract already exists at that address)

Reply:
But 5 lines later it does the check? https://github.com/gnosis/safe-contracts/blob/development/contracts/proxies/GnosisSafeProxyFactory.sol#L53

Either way, I added it here too